crates/javy/src/apis/json.rs

```diff
@@ -74,46 +93,77 @@ fn register<'js>(this: Ctx<'js>) -> Result<()> {
     Ok(())
 }
 
-fn call_json_parse(args: Args<'_>) -> Result<Value> {
+fn call_json_parse<'a>(args: Args<'a>, default: Function<'a>) -> Result<Value<'a>> {
     let (this, args) = args.release();
 
     match args.len() {
-        0 => Err(anyhow!("Error: \"undefined\" is not valid JSON")),
+        0 => Err(anyhow!(Exception::throw_syntax(
+            &this,
+            "undefined\" is not valid JSON"
+        ))),
         1 => {
             let val = args[0].clone();
             // Fast path. Number and null are treated as identity.
             if val.is_number() || val.is_null() {
                 return Ok(val);
             }
 
+            if val.is_symbol() {
+                return Err(anyhow!(Exception::throw_type(
+                    &this,
+                    "Expected string primitive"
+                )));
+            }
+
             let mut string = val_to_string(this.clone(), args[0].clone())?;
             let bytes = unsafe { string.as_bytes_mut() };
-            json::parse(this, bytes)
+            json::parse(this.clone(), bytes).map_err(|original| {
+                if original.downcast_ref::<SError>().is_none() {
+                    return original;
+                }
+
+                // TODO: Use the actual result here.
+                let _e = original.downcast::<SError>().unwrap();
```

Comment:
Why are we doing this?

Reply:
I forgot to use the actual error. For completeness, the spec tests don't care about the error message, they assert against the error type.